Assets/Scripts/PlayerShooting.cs

```diff
@@ -16,28 +22,26 @@ void Update()
 
     void FireProjectile()
     {
-        if (!projectilePrefab)
-        {
-            Debug.LogWarning($"Cannot fire projectile: {nameof(projectilePrefab)} is unset.", this);
-            return;
-        }
-
         if (!fireFrom)
         {
             Debug.LogWarning($"Cannot fire projectile: {nameof(fireFrom)} is unset.", this);
             return;
         }
 
-        var clone = Instantiate(projectilePrefab, fireFrom.position, fireFrom.rotation);
-        var body = clone.GetComponentInChildren<Rigidbody>();
-
-        if (!body)
+        if (!pickupAmmunition.ammunition)
         {
-            Debug.LogWarning("Projectile prefab does not have a RigidBody.", this);
+            return;
         }
-        else
+
+        var ammoBody = pickupAmmunition.ammunition.GetComponent<Rigidbody>();
+        if (!ammoBody)
         {
-            body.AddForce(fireFrom.forward * fireForce);
+            Debug.LogWarning("Projectile does not have a Rigidbody.", this);
+            return;
         }
```

Comment:
`Ammunition` component already has a reference to its Rigidbody. Think you can take use of that here.